docs/v1beta1-checklist-developer-guide.md

```diff
@@ -0,0 +1,128 @@
+   the conventions of [Kubernetes API design specifications] and the Crossplane
+   [API patterns design doc].
+3. Client: a set of functions that translate the Crossplane API to the cloud
+   provider API. Handles operations such as late initialization and comparison
+   between the two APIs.
+4. Controller: uses [`crossplane-runtime`] library to provide custom CRUD
+   operations to a shared reconciler pattern that is used across all Crossplane
+   API types. Calls upon the client to translate the objects it receives from
+   the cloud provider API and the Kubernetes API server.
+
+## Cloud Provider API Type
+
+As mentioned previously, the cloud provider API type is external to Crossplane
+and cannot be modified. However, it will inform how all other components are
+implemented, so it is important to have a strong understanding of both the
+fields it exposes and how they are used. Many cloud provider's will use default
+values for omitted fields that may not be intuitive or equal to the [zero-value]
+of the Go type they are implemented in by the SDK. It is usually not enough to
+just read the code exposed by the API. Authors of `beta` Crossplane APIs should
+be familiar with how the cloud provider responds to and modifies each field
+either by reading additional documentation or testing different configurations
+directly against the API.
+
+## Crossplane API Type
+
+The Crossplane API type is a Kubernetes representation of the cloud provider
+API. All `beta` Crossplane API types should be a [high fidelity] representation
+of the cloud provider API type. This means that any functionality that can be
+achieved using the cloud provider API should also be possible by creating the
+Crossplane representation of that API type. For full detail about how to design
+a Crossplane API type, please see our [API patterns design doc].
+
+- [ ] Any fields related to naming the external resource have been removed from
+  `spec.ForProvider`. They should be handled by the `crossplane-runtime`
+  [external name annotation].
+- [ ] All fields that can be configured at any point in the lifetime of the
+  external resource have been added to the `spec.forProvider` struct of the
+  Crossplane API type.
+- [ ] All fields that involve creating another external resource type that is
+  also represented by a separate Crossplane API type have been removed from
+  `spec.ForProvider`.
+- [ ] All fields that cannot be configured at any point in the lifetime of the
+  external resource have been added to the `status.atProvider` struct of the
+  Crossplane API type.
+- [ ] There is a code comment on every field that is descriptive of its
+  functionality.
+- [ ] All *optional* fields in `spec.forProvider` have been denoted with a `//
+  +optional` tag and are a pointer type.
+- [ ] All fields in `spec.forProvider` that cannot be modified after initial
+  external resource creation have been denoted with a `// +immutable` tag.
```

Comment:
`that cannot be modified after initial external resource creation` should be `that cannot be modified after it's assigned a value` because there are optional fields that you fill later but once you do, you cannot change them.